src/corehost/cli/fxr/fx_ver.cpp

```diff
@@ -108,20 +115,125 @@ int fx_ver_t::compare(const fx_ver_t&a, const fx_ver_t& b)
         return (a.m_patch > b.m_patch) ? 1 : -1;
     }
 
-    if (a.m_pre.empty() != b.m_pre.empty())
+    if (a.m_pre.empty() || b.m_pre.empty())
     {
-        // Either a is empty or b is empty
-        return a.m_pre.empty() ? 1 : -1;
+        // Either a is empty or b is empty or both are empty
+        return a.m_pre.empty() ? !b.m_pre.empty() : -1;
     }
 
-    // Either both are empty or both are non-empty (may be equal)
-    int pre_cmp = a.m_pre.compare(b.m_pre);
-    if (pre_cmp != 0)
+    // Both are non-empty (may be equal)
+    size_t idStart = 1;
+    for (size_t i = idStart; true; ++i)
     {
-        return pre_cmp;
+        if (a.m_pre[i] != b.m_pre[i])
+        {
+            // Found first character with a difference
+            if (a.m_pre[i] == 0 && b.m_pre[i] == _X('.'))
+            {
+                // identifiers both complete, b has an additional ident
```

Comment:
Nit: I would prefer to not use abbreviations like `ident` in comments... it's confusing for non-native speakers (i.e. me :-))